packages/flutter/lib/src/material/search_anchor.dart

```diff
@@ -907,11 +945,19 @@ class _SearchAnchorWithSearchBar extends SearchAnchor {
     super.isFullScreen,
     super.searchController,
     super.textCapitalization,
+    ValueChanged<String>? viewOnChanged,
+    ValueChanged<String>? viewOnSubmitted,
     required super.suggestionsBuilder
   }) : super(
     viewHintText: viewHintText ?? barHintText,
     headerTextStyle: viewHeaderTextStyle,
     headerHintStyle: viewHeaderHintStyle,
+    viewOnChanged: (String value) {
+      viewOnChanged?.call(value);
+    },
```

Comment:
Using a tear-off here produces a cleaner implementation. 
```suggestion
    viewOnChanged: viewOnChanged,
```